### ISSUE TITLE
Parsing ambiguity related to records.

### ISSUE THREAD

**Issue by cgrand** [dart-lang/sdk#52904]
* Dart SDK Version (`dart --version`)
3.0.5

This program doesn't compile: 
```dart
void main() {
  final x = 42;
  (x) >= 0; // ok
  (x) > 0; // ko
  (x) >> 0; // ko
}
```

Error:
```
A record type with exactly one positional field requires a trailing comma.
```

It seems to interpret `(x)` (before `>>` or `>`) as some kind of 1-item record type.

(This is not handwritten code but code generated by ClojureDart.)

**Comment by lrhn** [comment]
Looks like something that should work. Possibly the parser committed to an interpretation of `(x)` being a type a little too soon.

**Comment by jensjoha** [comment]
Duplicate of https://github.com/dart-lang/sdk/issues/52365.
@johnniwinther as I recall we talked about if the fix should be cherry-picked and decided it wasn't necessary. Is this still the thouht?

**Comment by johnniwinther** [comment]
The fix is in the current beta so I don't think it is necessary.

@cgrand Are you blocked on this or do you have a work-around?

**Comment by cgrand** [comment]
@johnniwinther I have a workaround
